bundles/binding/org.openhab.binding.lightwaverf/src/main/java/org/openhab/binding/lightwaverf/internal/LightwaveRFReceiver.java

```diff
@@ -0,0 +1,225 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+  * @author Neil Renaud
+  * @since 1.7.0
+  */
+public class LightwaveRFReceiver implements Runnable {
+    private static final Logger logger = LoggerFactory.getLogger(LightwaveRFReceiver.class);
+    private static final int DELAY_BETWEEN_RECEIVES_MS = 10;
+    
+    private final CopyOnWriteArrayList<LightwaveRFMessageListener> listeners = new CopyOnWriteArrayList<LightwaveRFMessageListener>();
+    private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(1);
+    
+    private final LightwaverfConvertor messageConvertor;
+    private final DatagramSocket receiveSocket;
+    private final Set<InetAddress> localIps;
+    
+    
+    private boolean running = false;
+    
+    public LightwaveRFReceiver(LightwaverfConvertor messageConvertor, int port) throws SocketException {
+    	this.messageConvertor = messageConvertor;
+    	this.receiveSocket = new DatagramSocket(port);
+    	localIps = getIpAddresses();
+    }
+    
+    /**
+     * Gets a list of INetAddresses for this server. Allowing us to filter out broadcast packets 
+     * we receive that we originally sent 
+     * @return
+     */
+    private Set<InetAddress> getIpAddresses(){
+    	Set<InetAddress> ips = new LinkedHashSet<InetAddress>();
+        try {
+            Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+            while (interfaces.hasMoreElements()) {
+                NetworkInterface iface = interfaces.nextElement();
+                // filters out 127.0.0.1 and inactive interfaces
+                if (iface.isLoopback() || !iface.isUp())
+                    continue;
+
+                Enumeration<InetAddress> addresses = iface.getInetAddresses();
+                while(addresses.hasMoreElements()) {
+                    InetAddress addr = addresses.nextElement();
+                    ips.add(addr);
+                }
+            }
+        } catch (SocketException e) {
+            throw new RuntimeException(e);
+        }
+        return ips;
+    }
+    	
+    /**
+     * Start the LightwaveRFReceiver
+     * Will set running true, initialise the socket and start the thread.
+     */
+    public synchronized void start() {
+        logger.info("Starting LightwaveRFReceiver");
+        running = true;
+        executor.scheduleWithFixedDelay(this, 0, DELAY_BETWEEN_RECEIVES_MS, TimeUnit.MILLISECONDS);
+    }
+    /**
+    * Stop the LightwaveRFSender
+    * Will close the socket wait for the thread to exit and null the socket
+    */
+    public synchronized void stop() {
+        logger.info("Stopping LightwaveRFReceiver");
+        running = false;
+        executor.shutdownNow();
+        receiveSocket.close();
+        logger.info("LightwaveRFReceiver Stopped");
+    }
+
+    /**
+    * Run method, this will listen to the socket and receive messages.
+    * The blocking is stopped when the socket is closed.
```

Comment:
You don't need a while loop with a scheduled executor that is set to run with a fixed delay. The executor will automatically re-schedule the thread for you after a fixed delay. This is why I prefer the scheduled thread executor with a single thread over creating a thread with a while loop that you have to be able to interrupt correctly, handle errors etc. The thread pool does this all for you out of the box and makes code much cleaner.